doc/sphinx/source/dbk.rst

```diff
@@ -34,3 +34,58 @@ buffer are as follows: point (0,0) is the top left of the image and each pixel c
 padding. So to get a pixel value from a certain row, add an offset of row_index * width * 3 to the index. The fourcc
 code that comes closest to this is the BI_RGB 24bpp format. Check the `include/CL/cl_exp_defined_builtin_kernels.h` for
 info on DBK attributes and `tests/runtime/test_dbk_jpeg.c` for an example program making use of the JPEG DBKs.
+
+
+exp_onnx_inference
+__________________
+
+This experimental DBK provides the ability to ingest an ONNX file and perform
+inference with the contained network.
+
+Building with ONNX kernel support
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In order to support this builtin kernel through the CPU devices, PoCL needs to be built with ONNX Runtime
+support.
+
+
+Distro-provided ONNX Runtime
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you are on a Linux distribution that packages ONNX Runtime, install the
+development package for it and build PoCL normally. The ONNX Runtime should be
+detected automatically if it is a sufficiently recent version that ships with
+CMake configuration files.
+
+Building with a custom ONNX Runtime
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Some popular Linux distributions don't provide prebuilt packages of ONNX Runtime
+so you may have to
+`build your own <https://onnxruntime.ai/docs/build/inferencing.html>` and tell
+CMake where to find it. Steps for building a suitable copy of ONNX Runtime (run
+outside of the PoCL source tree) are as follows:
+
+    git clone --recursive https://github.com/microsoft/onnxruntime/
+    cd onnxruntime
+    ./build.sh --config Release --build_shared_lib --parallel --compile_no_warning_as_error --skip_submodule_sync --skip_tests --cmake_extra_defines CMAKE_INSTALL_PREFIX=${HOME}/onnxruntime-bin
```

Comment:
What does `--skip_submodule_sync` do?

Reply:
It's part of the recommended flags from the build instructions on the onnxruntime website. `./build.sh --help` clarifies that:
```
  --skip_submodule_sync
                        Don't do a 'git submodule update'. Makes the Update phase faster.
```

Which makes sense after cloning the repo with `--recursive` right on the line above (as that clones and updates all submodules). Dropping the `--recursive` flag from the clone and the `--skip_submodule_sync` from the build flags is also an option.